pkg/controller/clone-controller.go

```diff
@@ -185,11 +185,9 @@ func (r *CloneReconciler) Reconcile(_ context.Context, req reconcile.Request) (r
 			"NOT has annotation(AnnCloneOf)", !metav1.HasAnnotation(pvc.ObjectMeta, AnnCloneOf),
 			"isBound", isBound(pvc, log),
 			"has finalizer?", HasFinalizer(pvc, cloneSourcePodFinalizer))
-		if HasFinalizer(pvc, cloneSourcePodFinalizer) {
-			// Clone completed, remove source pod and finalizer.
-			if err := r.cleanup(pvc, log); err != nil {
-				return reconcile.Result{}, err
-			}
+		// Clone completed, remove source pod and/or finalizer
+		if err := r.cleanup(pvc, log); err != nil {
```

Comment:
now cleanup will be called very frequently, basically whenever the status of any PVC changes.  And I believe the list call is pretty expensive (not in memory) so I think we should find a way to minimize the frequency of cleanup calls.